### DISCUSSION TITLE
Registration with social-app-django

### DISCUSSION THREAD

**Discussion by lavisco** [python-social-auth/social-core#1046] [Q&A]
### Expected behaviour

Is there any way to implement social-app-django for user registration? If yes then please someone guide me how to do this. Any help will be appreciated.

### Actual behaviour

It only allows for user login and not for sign up

**Comment by jfschaff** [comment]
I just added this app to a Django project, and my understanding is that the same process is used both for registration and login. But this can be configured depending on what you want to do by changing the "pipeline" (see doc). The default pipeline does both registration and login.

The default behavior I get is the following: using a slightly modified OpenID Connect provider, and the default pipeline :
- on first social login by a user unknown to the Django user base, a new Django user is created and associated with the OpenID user provided by the identify server;
- on next login, the same user is recognized and just authenticated.

But this project is very modular and can be modified by changing:
- the pipeline (I think this is the first thing you want to have a closer look at)
- the provider (OAuth, OpenID, etc.)
- the strategy
- etc.

I also had to modify the app, as described in python-social-auth/social-app-django#323, but this is not 100% clear yet.

**Comment by tiago-peres** [comment]
> ### Actual behaviour
> It only allows for user login and not for sign up

Not true.

**Comment by nijel** [comment]
Having `social_core.pipeline.user.create_user` in the pipeline should be enough for that.
